controllers/space/MembershipController.php

```diff
@@ -68,4 +69,51 @@ public function actionIndex($spaceId){
         }
         return $this->returnPagination($query,$pagination,$results);
     }
+
+    /**
+     * @return void
+     * Get all the spaces list of the user
+     */
+    //Get the list of all spaces of the user (issue-8 new endpoint)
+    public function actionGetSpaces(){
+       $userId = Yii::$app->user->id;
+
+       $user = User::findOne($userId);
+
+       if($user == null){
+           return $this->returnError(404, 'User not found!');
+       }
+
+       $spaceMemberships = Membership::find()->where(['user_id' => $userId]);
+
+        if($spaceMemberships->all() == null){
+            return $this->returnError(404, 'Space not found for the user : '.$userId);
+        }
+
+        $results = [];
+        $pagination = $this->handlePagination($spaceMemberships);
+
+        foreach ($spaceMemberships->all() as $spaceMembership){
+            $results[] = $this->getSpacesListData($spaceMembership);
+        }
+
+        return $this->returnPagination($spaceMemberships,$pagination,$results);
+    }
+
+    public function getSpacesListData(Membership $membership){
+        return [
+            'user' => UserDefinitions::getUserShort($membership->user),
+            'space_id' => $membership->space_id,
+            'role' => $membership->group_id,
+            'status' => $membership->status,
+            'can_cancel_membership' => $membership->can_cancel_membership,
+            'send_notifications' => $membership->send_notifications,
+            'show_at_dashboard' => $membership->show_at_dashboard,
+            'originator_user' => ($membership->originator !== null) ? UserDefinitions::getUserShort($membership->originator) : null,
```

Comment:
what is the `originator_user`?

Reply:
originator_user = owner of that space

Reply:
it was `null` in my test. if we will have the `owner` key, i suggest to remove the `originator_user` or what do you say?

Reply:
Hi @donni106 
Actually, it was null because the user is not the owner of that space. 

Yes, we can remove this originator_user if you want?

Reply:
yes please. having this data only if the user is owner will not help us i guess.
having the `owner` key always will hold the information at any time.